### PR TITLE
WIP CamelSource e2e test

### LINKED ISSUE #271: e2e test for camel-k source
## Expected Behavior

have a e2e test for the camel-k source

## Actual Behavior

no test for this

### PR DESCRIPTION
**Pull request by maschmid**
Fixes #271

## Proposed Changes

  * Makes CamelSource OwnerRefable 
  * Adds camelk deployment to e2e test setup
  * Adds TestCamelSource

**Release Note**

NONE

### CODE REVIEW

File: test/e2e/camel_source_test.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+	t.Logf("Waiting for all resources ready")
+	client.WaitForAllTestResourcesReadyOrFail()
+
+	t.Logf("Sleeping for 3s to let the timer tick at least once")
+	time.Sleep(3 * time.Second)
+
+	if err := client.CheckLog(loggerPodName, common.CheckerContains(body)); err != nil {
+		t.Fatalf("Strings %q not found in logs of logger pod %q: %v", body, loggerPodName, err)
+	}
+}
```
Comment (mattmoor) [inline test/e2e/camel_source_test.go]: Add trailing newline:
```suggestion
}

```

File: test/e2e/camel_source_test.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
+//+build e2e
+
+/*
+Copyright 2019 The Knative Authors
```
Comment (mattmoor) [inline test/e2e/camel_source_test.go]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

**Comment by maschmid** [comment]
Still WIP

Cannot work now in master, as CamelSource is broken ATM

Also, right now the 100-camelk-operator-1.0.0.RC1.yaml is just the collection of all the yamls from https://github.com/apache/camel-k/tree/1.0.0-RC1/deploy  , and it will need some tuning with role bindings to make work for any test namespace.

**Comment by maschmid** [comment]
pull-knative-eventing-contrib-integration-tests fails because of https://github.com/knative/eventing-contrib/issues/805

> camel-controller-manager-0                                       0/1     CrashLoopBackOff

**Comment by maschmid** [comment]
Still needs to setup registry for the camel-k builds per https://camel.apache.org/camel-k/latest/installation/registry/gcr.html

**Comment by matzew** [comment]
@nicolaferraro any help here ?

**Comment by maschmid** [comment]
Superseded by  https://github.com/knative/eventing-contrib/pull/892

**Pull request closed without merging**